app/src/main/java/com/kms/cura/view/activity/BookAppointmentActivity.java

```diff
@@ -0,0 +1,204 @@
+                (ArrayList) apptLength, apptLengthSelected);
+        spnApptLength.setAdapter(adapterApptLength);
+        spnApptLength.setSelection(adapterApptLength.getCount());
+    }
+
+    public void initSpinner() {
+        spnApptLength = (Spinner) findViewById(R.id.spnAppLength);
+        spnFacilities = (Spinner) findViewById(R.id.spnFacility);
+    }
+
+    public void setUpData() {
+        hintText = getResources().getString(R.string.HintText);
+        String[] length = getResources().getStringArray(R.array.ApptLength);
+        apptLength = new ArrayList<>();
+        Collections.addAll(apptLength, length);
+        apptLength.add(hintText);
+        doctorFacilityName = new ArrayList<>();
+        doctorFacilityName.add("A");
+        doctorFacilityName.add("B");
+        doctorFacilityName.add("C");
+        doctorFacilityName.add(hintText);
+    }
+
+    public void initTextView() {
+        txtDate = (TextView) findViewById(R.id.txtDate);
+        txtTime = (TextView) findViewById(R.id.txtTime);
+        edtComment = (EditText) findViewById(R.id.edtComment);
+    }
+
+    @Override
+    public void onClick(View v) {
+        int id = v.getId();
+        if (id == R.id.btnChooseDate) {
+            chooseDate();
+        } else if (id == R.id.btnChoseTime) {
+            chooseTime();
+        }
+        else if(id == R.id.btnBack){
+            finish();
+        }
+
+    }
+
+    private void chooseDate(){
+        setCurrentDate();
+        Dialog dateDialog = new DatePickerDialog(this, AlertDialog.THEME_HOLO_LIGHT, myDateListener, year, month - 1, day);
+        dateDialog.show();
+    }
+
+    private void chooseTime(){
+        if(spnApptLength.getSelectedItemPosition()!= apptLength.size()-1 &&
+                spnFacilities.getSelectedItemPosition() != doctorFacilityName.size() -1 && selectDate ) {
+            Intent toChooseTime = new Intent(this, ChooseTimeActivity.class);
+            Bundle bundle = createBundle();
+            toChooseTime.putExtras(bundle);
+            startActivityForResult(toChooseTime, REQUEST_cODE);
+        }
+        else {
+            Toast.makeText(this, getResources().getString(R.string.BookAptError), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private DatePickerDialog.OnDateSetListener myDateListener = new DatePickerDialog.OnDateSetListener() {
+        @Override
+        public void onDateSet(DatePicker datePicker, int year, int month, int day) {
+            setSpecificDate(day, month + 1, year);
+        }
+    };
+
+    private void setCurrentDate() {
+        calendar = Calendar.getInstance();
+        day = calendar.get(Calendar.DAY_OF_MONTH);
+        month = calendar.get(Calendar.MONTH) + 1;
+        year = calendar.get(Calendar.YEAR);
+
+    }
+
+    private void setSpecificDate(int day, int month, int year) {
+        if(day < this.day || month < this.month || year < this.year){
```

Comment:
So for example, if today is ( 30, 1, 2016 ) and the input date is ( 1, 2, 2016 ), what will be the output ?